packages/vite/src/node/plugins/resolve.ts

```diff
@@ -288,8 +294,6 @@ function tryResolveFile(
       }
       const index = tryFsResolve(file + '/index', options)
       if (index) return index + postfix
-    } else {
-      return normalizePath(ensureVolumeInPath(file)) + postfix
```

Comment:
@Timsonrobl is it necessary to remove this branch? Could you add a comment where that branch was explaining why it is not needed in that case?

Reply:
It is necessary to either remove it or add another parameter and condition `else if (tryDirectory)` otherwise `dont add extension to directory name (./dir-with-ext.js/index.js)` test included in this commit would not pass (and not fail, it would hang test job instead).
I wasn't able to figure out why that branch existed. I checked that it was never reached during `yarn test`, and in my small vue 3 project. Also looking [here](https://nodejs.org/api/esm.html#esm_resolver_algorithm) I don't understand why resolver would ever need to return directory path. That doesn't mean that reason don't exist 😅

Reply:
I just went back in time, and I actually touched this particular line in a commit. You can see how that function was back then here https://github.com/patak-js/vite/blob/844a3820f969f3869295a6c6477085f2899e26ba/packages/vite/src/node/plugins/resolve.ts#L246

At that point, that was the line that was reached when it wasn't a directory.

Looks like this is the commit where that changed, because `tryIndex` was moved to the dir check: https://github.com/vitejs/vite/commit/c741872e6ca975f507ec89581b742a1da19a0cb0, so when tryIndex is false, if it is a dir, it started to hit that line. Later a check for the case of a file was added at the start of the function, so looks like this is indeed a dead branch due to the evolution of the code. @Timsonrobl you can review this change set if you would like but this PR looks fine to me then.